src/hypothesis/extra/django/models.py

```diff
@@ -17,159 +17,402 @@
+            if field.choices:
+                choices = [
+                    value
+                    for (value, name)
+                    in field.choices
+                ]
+                if field.blank:
+                    choices.extend(blank_values)
+                strategy = st.sampled_from(choices)
+            else:
+                # Run the factory function.
+                strategy = func(field, **kwargs)
+            # Add in null values.
+            if field.null:
+                strategy = st.one_of(st.none(), strategy)
+            # Filter by validators.
+            strategy = strategy.filter(validator_to_filter(field))
+            return strategy
+        return create_field_strategy
+    return decorator
+
+
+def _simple_field_strategy(blank_values=()):
+    def decorator(func, **defaults):
+        @defines_field_strategy(blank_values=blank_values)
+        def create_simple_field_strategy(field, **kwargs):
+            params = defaults.copy()
+            params.update(kwargs)
+            return func(**params)
+        return create_simple_field_strategy
+    return decorator
+
+
+def _fake_factory_field_strategy(blank_values=()):
+    def decorator(strategy_name):
+        @defines_field_strategy(blank_values=blank_values)
+        def create_fake_factory_field_strategy(field, min_size=None,
+                                               max_size=None):
+            strategy = fake_factory(strategy_name)
+            # Add in blank values.
+            if field.blank:
+                strategy = st.one_of(strategy, *blank_values)
+            # Emulate min size.
+            if min_size is not None:
+                strategy = strategy.filter(lambda v: len(v) >= min_size)
+            # Emulate max size.
+            if max_size is not None:
+                strategy = strategy.filter(lambda v: len(v) <= max_size)
+            # Emulate blank.
+            if field.blank and blank_values:
+                strategy = st.one_of(st.sampled_from(blank_values), strategy)
+            # All done!
+            return strategy
+        return create_fake_factory_field_strategy
+    return decorator
+
+
+big_integer_field_values = _simple_field_strategy()(
+    st.integers,
+    min_value=-9223372036854775808,
+    max_value=9223372036854775807,
+)
+add_default_field_mapping(dm.BigIntegerField, big_integer_field_values)
+
+
+binary_field_values = _simple_field_strategy()(st.binary)
+add_default_field_mapping(dm.BinaryField, binary_field_values)
+
+
+boolean_field_values = _simple_field_strategy()(st.booleans)
+add_default_field_mapping(dm.BooleanField, boolean_field_values)
+
+
+@add_default_field_mapping(dm.CharField)
+@add_default_field_mapping(dm.TextField)
+@defines_field_strategy(blank_values=('',))
+def char_field_values(field, **kwargs):
+    """A strategy of valid values for the given CharField."""
+    if field.blank:
+        kwargs.setdefault('min_size', 0)
+    if field.max_length:
```

Comment:
Nit: is it possible to have a field with `max_length == 0`? Maybe update this to `if field.max_length is not None:`